### PR TITLE
React/Flask: Added sequencing ID field in Dataset table

### PR DESCRIPTION
**Pull request by jennyziyi-xu**
Added a column `sequencing_id` to the dataset table. The column can be sorted and filtered. Added test cases in test_datasets.py. Added field `sequencing_id` to the global test cases in conftest.py. 

Comment: Currently, only 4 datasets for pytest are documented in flask.md, should we update flask.md? There are 5 datasets in conftest.py.

### CODE REVIEW

**Comment by hannahle** [comment]
> Comment: Currently, only 4 datasets for pytest are documented in flask.md, should we update flask.md? There are 5 datasets in conftest.py.

This is because I updated conftest.py recently. I will update the flask.md. Thanks for pointing that out!

File: flask/app/datasets.py [hunk 1]
```diff
@@ -94,6 +95,8 @@ def list_datasets(page: int, limit: int) -> Response:
         # Since this is an elif clause, we know special cases are already handled above.
         # order_by has been restricted to a known list, so we can safely use getattr
         order = getattr(models.Dataset, order_by)
+    elif order_by == "sequencing_id":
+        order = models.Dataset.sequencing_id
```
Comment (kevinlul) [inline flask/app/datasets.py:99]: This is a field of the dataset table, so it is already handled by the prior elif by being included in the list and does not need to be explicitly done here.

File: flask/tests/conftest.py [hunk 2]
```diff
@@ -218,6 +218,7 @@ def test_database(client):
         condition=DatasetCondition.Somatic,
         created_by_id=admin.user_id,
         updated_by_id=admin.user_id,
+        sequencing_id=5,
```
Comment (kevinlul) [inline flask/tests/conftest.py:221]: This column is a string, so it shouldn't be an int here

**Review by kevinlul — CHANGES_REQUESTED** [review]
Please enter a pull request title instead of the automatically-generated one!

**Comment by kevinlul** [comment]
> Please enter a pull request title instead of the automatically-generated one!

**Comment by jennyziyi-xu** [comment]
> Please enter a pull request title instead of the automatically-generated one!

Done. Sorry, was a bit confused.

**Review by kevinlul — APPROVED** [review]
LGTM!

**Pull request merged**